app/services/remove_status_service.rb

```diff
@@ -67,31 +70,35 @@ def remove_from_lists
     end
   end
 
-  def remove_from_affected
-    @mentions.map(&:account).select(&:local?).each do |account|
-      redis.publish("timeline:#{account.id}", @payload)
+  def remove_from_mentions
+    # For limited visibility statuses, the mentions that determine
+    # who receives them in their home feed are a subset of followers
+    # and therefore the delete is already handled by sending it to all
+    # followers. Here we send a delete to actively mentioned accounts
+    # that may not follow the account
+
+    @status.active_mentions.find_each do |mention|
+      redis.publish("timeline:#{mention.account_id}", @payload)
     end
```

Comment:
I understand this is to avoid fetching account information from the database, but this will send events about remote accounts to redis. Is it worth it? Does it make sense?

Reply:
You’re right... but publishing to a channel with no subscribers should be O(0) so probably fine this way.